### DISCUSSION TITLE
First generation cards prior to RC-S833

### DISCUSSION THREAD

**Discussion by hanpuliu-charles** [nullableVoidPtr/nfcf-info#3] [Analysis]
Hi, there. I am check to see if you have any infomation on what could be the first generation card that was mass produced, used for Octopus system when it launched. Several reasons on why I believe that this is a different series from RC-S833 series of IC. First, it seems like these cards are unable to be detected with mobile phone nfc, could suggest that it does not respond to wildcard polling(I have to experiment with it more to see if this is truly the reason, like trying to poll with octopus system code). Next, I have found an [article](http://www.9to.com/Article/show-4446.html) from 2006 talking about interoperability between Octopus card and then Shenzhentong, about how older octopus cards used something called SONYSEAC card as opposed to S833 card used by Shenzhentong. I would like to check if you have encountered this family of card in your research before?

**Comment by hanpuliu-charles** [comment]
I also found mention of SEAC [here](https://www.octopus.com.hk/sc/document/door_access_unit_residential.pdf). So from what is mentioned, it looks like it was a seperate encryption scheme from the later DES cards.

Edit: mention [here](https://lihkg.com/thread/3325200/page/18) too. So this could be the reason why it isn't supported by NFC on mobile phone.

[FCC document for S480C](https://fccid.io/AK8S480C/User-Manual/User-Manual-86124) says that SEAC is Sony Encryption Algorithm for Cards.

**Comment by nullableVoidPtr** [comment]
It has been already listed under [RC-S830](https://nullablevoidptr.github.io/nfcf-info/IC-chips/card/#rc-s830), with some detail, but the additional references are greatly appreciated. There is also another reference to SEAC/Octopus under the old Proxmark forums [here](https://www.proxmark.io/www.proxmark.org/forum/viewtopic.php%3Fid=1446.html)

My possible theories are that Sony Encryption Algorithm for Cards (SEAC) is either: 
* a modification of single DES for longer 
* a reduced variant of [CLEFIA](https://www.sony.net/Products/cryptography/clefia/), a Feistel network first published in 2007 but with 128-bit block size; or even more unlikely,
* an early variant of [Piccolo](https://www.iacr.org/archive/ches2011/69170343/69170343.pdf), another Feistel network with 64-bit block size but 80-bit keys.

If it's a entirely proprietary cipher which is highly likely, I further suspect SEAC has the following properties:
* lightweight cipher with some Feistel network structure
  * simple to implement in embedded 8-bit RISC CPUs like the one specified for RC-S830, either a Fujitsu F2MC-8FX core or Panasonic AM core
  * hence, potentially weak under cryptanalysis or specialized bruteforce!
  * use of Feistel networks follows Sony's literature and also DES's structure.
* 64-bit key size and block size
  * implied in the FCC Users' Manual, as DES itself has 56-bit keys
* *maybe* has a trade-off between longer key-length and simpler round function, or less likely key scheduling, compared to DES

Of course, it's hard to know for sure without proper reverse engineering. I'll look to incorporate this information in the coming days crediting you.

**Comment by hanpuliu-charles** [comment]
Thanks for providing some guesses as to what it is. I actually have an RC-S833, and it can be read by NFC devices, it shows up with taginfo. So this might be something else or different firmware on the card resulting in different behaviors. I am just surprised at how there are so much research work on crypto-1 attacks and reverse engineering, but there barely seems to be any on Sony's product even though they are both proprietary.
